scaler/handlers.go

```diff
@@ -144,32 +145,32 @@ func (e *impl) GetMetricSpec(
 	}
 
 	var targetPendingRequests int64
-	var hostMetricSpec []*externalscaler.MetricSpec
-	for _, host := range hosts {
-		if host == interceptor {
-			targetPendingRequests = e.targetMetricInterceptor
-		} else {
-			target, err := e.routingTable.Lookup(host)
-			if err != nil {
-				lggr.Error(
-					err,
-					"error getting target for host",
-					"host",
-					host,
-				)
-				return nil, err
-			}
-			targetPendingRequests = int64(target.TargetPendingRequests)
+	var host = hosts[0] // We are only interested in the first host to get the targetPendingRequests
+
+	if host == interceptor {
+		targetPendingRequests = e.targetMetricInterceptor
+	} else {
+		target, err := e.routingTable.Lookup(host)
+		if err != nil {
+			lggr.Error(
+				err,
+				"error getting target for host",
+				"host",
+				host,
+			)
+			return nil, err
 		}
+		targetPendingRequests = int64(target.TargetPendingRequests)
+	}
 
-		hostMetricSpec = append(hostMetricSpec, &externalscaler.MetricSpec{
-			MetricName: host,
+	metricSpecs := []*externalscaler.MetricSpec{
+		{
+			MetricName: strings.Join(hosts, ","),
```

Comment:
Be careful here because metric name has max length of 64 characters if I'm not wrong.  Although, if that limit doesn't apply here, multiple hosts will generate a not usable metric name due to the length.
WDYT if we return something fixed like `http-requests`? Also, we support other ScaledObject triggers in the future, this will make it more readable between other metric names (but this isn't a huge reason to go atm)